systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+
+            assert_about_eq!(
+                tester.get_egt_warning_temperature().get::<degree_celsius>(),
+                900.
+            );
+        }
+
+        #[test]
+        fn max_starting_egt_at_or_above_25000_feet_is_982_degrees() {
+            let tester = tester_with()
+                .starting_apu()
+                .and()
+                .indicated_altitude(Length::new::<foot>(25000.))
+                .run(Duration::from_secs(1));
+
+            assert_about_eq!(
+                tester.get_egt_warning_temperature().get::<degree_celsius>(),
+                982.
+            );
+        }
+
+        #[test]
+        fn starting_apu_n_is_never_below_0() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(10));
+
+                assert!(tester.get_n().get::<percent>() >= 0.);
+
+                if tester.apu_is_available() {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn restarting_apu_which_is_cooling_down_does_not_suddenly_reduce_egt_to_ambient_temperature(
+        ) {
+            let mut tester = tester_with().cooling_down_apu();
+
+            assert!(tester.get_egt().get::<degree_celsius>() > 100.);
+
+            tester = tester
+                .then_continue_with()
+                .starting_apu()
+                .run(Duration::from_secs(5));
+
+            assert!(tester.get_egt().get::<degree_celsius>() > 100.);
+        }
+
+        #[test]
+        fn restarting_apu_which_is_cooling_down_does_reduce_towards_ambient_until_startup_egt_above_current_egt(
+        ) {
+            let mut tester = tester_with().cooling_down_apu();
+
+            let initial_egt = tester.get_egt();
+
+            tester = tester
+                .then_continue_with()
+                .starting_apu()
+                .run(Duration::from_secs(5));
+
+            assert!(tester.get_egt() < initial_egt);
+        }
+
+        #[test]
+        fn start_contactor_is_energised_when_starting_until_n_55() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                let n = tester.get_n().get::<percent>();
+
+                assert_eq!(tester.start_contactor_energized(), n < 55.);
+
+                if n == 100. {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn start_contactor_is_energised_when_starting_until_n_55_even_if_master_sw_turned_off() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                let n = tester.get_n().get::<percent>();
+
+                if n > 30. {
+                    tester = tester.master_off();
+                }
+
+                assert_eq!(tester.start_contactor_energized(), n < 55.);
```

Comment:
You could ensure that this condition hits both ways at least once.

Reply:
Given that the test won't stop until N = 100, I guess it does?

Reply:
I've added an assertion before the loop to ensure we check the < 55 case at least once.